lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -124,150 +124,128 @@ def run_module():
+                'storage_pool']
+            fail_with_reason(reason)
+
+        # Check if datacenter exists, it must in order to continue
+        datacenter_uuid = tacp_utils.get_component_fields_by_name(
+            module.params['datacenter'], 'datacenter', api_client)
+        if datacenter_uuid:
+            instance_params['datacenter_uuid'] = datacenter_uuid
         else:
-            # TODO - his will eventually need to be modified so that any
-            # boot order is possible for any extra NICs
-
-            vnic_uuid = str(uuid4())
-            vnic_boot_order = len(boot_order) + i
-
-            vnic_payload = tacp.ApiAddVnicPayload(
-                automatic_mac_address=automatic_mac_address,
+            # Datacenter does not exist - must fail the task
+            reason = "Datacenter %s does not exist, cannot continue." % module.params[
+                'datacenter']
+            fail_with_reason(reason)
+
+        # Check if migration_zone exists, it must in order to continue
+        migration_zone_uuid = tacp_utils.get_component_fields_by_name(
+            module.params['mz'], 'migration_zone', api_client)
+        if migration_zone_uuid:
+            instance_params['migration_zone_uuid'] = migration_zone_uuid
+        else:
+            # Migration zone does not exist - must fail the task
+            reason = "Migration Zone %s does not exist, cannot continue." % module.params[
+                'mz']
+            fail_with_reason(reason)
+
+        # Check if template exists, it must in order to continue
+        template_uuid = tacp_utils.get_component_fields_by_name(
+            module.params['template'], 'template', api_client)
+        if template_uuid:
+            instance_params['template_uuid'] = template_uuid
+            boot_order = tacp_utils.get_component_fields_by_name(
+                module.params['template'], 'template', api_client, fields=['name', 'uuid', 'bootOrder'])
+        else:
+            # Template does not exist - must fail the task
+            reason = "Template %s does not exist, cannot continue." % module.params[
+                'template']
+            fail_with_reason(reason)
+
+        network_payloads = []
+
+        for i, nic in enumerate(module.params['nics']):
+            if nic['type'].lower() == "vnet":
+                network_uuid = tacp_utils.get_component_fields_by_name(
+                    nic['network'], 'vnet', api_client)
+            elif nic['type'].lower() == "vlan":
+                network_uuid = tacp_utils.get_component_fields_by_name(
+                    nic['network'], 'vlan', api_client)
+
+            if 'mac_address' in nic.keys():
+                if nic['mac_address']:
+                    automatic_mac_address = False
+                    mac_address = nic['mac_address']
+            else:
+                automatic_mac_address = True
+                mac_address = None
```

Comment:
Just use the `get` api of the dict. It defaults to `None` if the key is not in the dict.
```
mac_address = nic.get('mac_address')
automatic_mac_address = not bool(mac_address)
```

Reply:
smart, changing this